src/main/kotlin/no/nav/syfo/oppfolgingstilfelle/domain/OppfolgingstilfelleDag.kt

```diff
@@ -0,0 +1,95 @@
+)
+
+fun List<OppfolgingstilfelleDag>.groupOppfolgingstilfelleList(): List<Oppfolgingstilfelle> {
+    val oppfolgingstilfelleList = ArrayList<Oppfolgingstilfelle>()
+    var oppfolgingstilfelleSykedagList = ArrayList<OppfolgingstilfelleDag>()
+    var notSykedagSinceLastSykedagCounter = 0
+
+    this.forEach {
+        when {
+            it.isArbeidsdag() -> {
+                notSykedagSinceLastSykedagCounter++
+            }
+
+            it.isFeriedag() -> {
+                if (notSykedagSinceLastSykedagCounter > 0) {
+                    // Only count Feriedag if at least one Arbeidsdag since last Sykedag
+                    notSykedagSinceLastSykedagCounter++
+                }
+            }
+            else -> { // isSykedag
+                oppfolgingstilfelleSykedagList.add(it)
+                notSykedagSinceLastSykedagCounter = 0
+            }
+        }
+
+        val noSykedagLast16days = notSykedagSinceLastSykedagCounter >= 16 && oppfolgingstilfelleSykedagList.isNotEmpty()
+        if (noSykedagLast16days) {
+            val newOppfolgingstilfelle = Oppfolgingstilfelle(
+                personIdentNumber = this.mapNotNull { dag -> dag.priorityOppfolgingstilfelleBit?.personIdentNumber }.first(),
+                start = oppfolgingstilfelleSykedagList.first().dag,
+                end = oppfolgingstilfelleSykedagList.last().dag,
+                virksomhetsnummerList = emptyList(),
+            )
+            oppfolgingstilfelleList.add(newOppfolgingstilfelle)
+
+            // Reset variables
+            oppfolgingstilfelleSykedagList = ArrayList()
+            notSykedagSinceLastSykedagCounter = 0
+        }
+    }
+
+    if (oppfolgingstilfelleSykedagList.isNotEmpty()) {
+        val virksomhetsnummerList: List<Virksomhetsnummer> = oppfolgingstilfelleSykedagList.mapNotNull {
+            it.priorityOppfolgingstilfelleBit?.virksomhetsnummer
+        }.distinct().map { virksomhetsnummer ->
+            Virksomhetsnummer(virksomhetsnummer)
+        }
+        val lastOppfolgingstilfelle = Oppfolgingstilfelle(
+            personIdentNumber = this.mapNotNull { dag -> dag.priorityOppfolgingstilfelleBit?.personIdentNumber }.first(),
+            start = oppfolgingstilfelleSykedagList.first().dag,
+            end = oppfolgingstilfelleSykedagList.last().dag,
+            virksomhetsnummerList = virksomhetsnummerList,
+        )
+        oppfolgingstilfelleList.add(lastOppfolgingstilfelle)
+    }
+
+    return oppfolgingstilfelleList
+}
+
+fun OppfolgingstilfelleDag.isArbeidsdag() =
+    priorityOppfolgingstilfelleBit
+        ?.tagList
+        ?.let {
+            it in (
+                (Tag.SYKMELDING and Tag.PERIODE and Tag.FULL_AKTIVITET)
+                    or (Tag.SYKEPENGESOKNAD and Tag.ARBEID_GJENNOPPTATT)
+                    or (Tag.SYKEPENGESOKNAD and Tag.BEHANDLINGSDAGER)
+                )
+        }
+        ?: true
+
+fun OppfolgingstilfelleDag.isFeriedag() =
+    priorityOppfolgingstilfelleBit
```

Comment:
Synes `priorityOppfolgingstilfelleBit != null && priorityOppfolgingstilfelleBit.tagList in (Tag.SYKEPENGESOKNAD and (Tag.FERIE or Tag.PERMISJON))` er litt mer lesbart her, men kanskje smag og behag.
Kan også vurdere tilsvarende over (`priorityOppfolgingstilfelleBit == null || priorityOppfolgingstilfelleBit.tagList in ... `)

Reply:
Er jo litt avhengig fra funskjon til funksjon, men her sjekkes kun en ting om innover i objektet og da synes jeg det blir veldig kort, konsist og lesbart med bruk av elvis-operatør.

Merker jo at det krever litt tilvenning å lese, men  jeg har begynt å like det etterhvert som jeg har brukt det mer.